app/handlers/testConfig.py

```diff
@@ -29,79 +29,121 @@ class TestConfigHandler(tornado.web.RequestHandler):
     @apiError error_message The error message to display.
     """
     def post(self):
-
         identifier = self.get_body_argument("identifier")
         test_flag = self.get_body_argument("test_flag")
         print test_flag
 
         frame_start = int(self.get_body_argument("frame_start", default = 0))
         num_frames = int(self.get_body_argument("num_frames", default = 120))
 
-        if test_flag == "feature":
-            print "running feature"
-            status_code, reason = self.runConfigTestFeature(identifier, frame_start, num_frames)
-        elif test_flag == "object":
-            print "running object"
-            status_code, reason = self.runConfigTestObject(identifier, frame_start, num_frames)
+        status_code, response_message = TestConfigHandler.handler(identifier, frame_start, num_frames, test_flag)
 
         if status_code == 200:
             self.finish("Testing tracking")
         else:
             raise tornado.web.HTTPError(reason=reason, status_code=status_code)
 
-    def runConfigTestFeature(self, identifier, frame_start, num_frames):
+    @staticmethod
+    def handler(identifier, frame_start, num_frames, test_flag):
         project_path = get_project_path(identifier)
         if not os.path.exists(project_path):
            return (500, 'Project directory does not exist. Check your identifier?')
 
+        if test_flag == "feature":
+            print "running feature"
+            TestConfigFeatureThread(identifier, frame_start, num_frames, TestConfigHandler.test_feature_callback).start()
+        elif test_flag == "object":
+            print "running object"
+            TestConfigObjectThread(identifier, frame_start, num_frames, TestConfigHandler.test_feature_callback).start()
+
```

Comment:
We need another check here incase they pass in an invalid flag, right now it would just return success.